src/Platform/Microsoft.Testing.Extensions.OpenTelemetry/OpenTelemetryProviderExtensions.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Testing.Extensions.OpenTelemetry;
+using Microsoft.Testing.Platform.Builder;
+using Microsoft.Testing.Platform.Services;
+
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace Microsoft.Testing.Extensions;
+
+/// <summary>
+/// Extensions for adding AppInsights telemetry provider.
+/// </summary>
+[Experimental("TPEXP", UrlFormat = "https://aka.ms/testingplatform/diagnostics#{0}")]
+public static class OpenTelemetryProviderExtensions
+{
+    /// <summary>
+    /// Adds OpenTelemetry tracing and metrics providers to the application builder, allowing customization of tracing
+    /// and metrics configuration.
+    /// </summary>
+    /// <remarks>This method enables distributed tracing and metrics collection for the application. To
+    /// customize telemetry behavior, provide configuration delegates for tracing and/or metrics. If no delegates are
+    /// supplied, default OpenTelemetry settings are used.</remarks>
+    /// <param name="builder">The application builder to which the OpenTelemetry providers will be added. Cannot be null.</param>
+    /// <param name="withTracing">An optional delegate to configure the tracing provider. If null, default tracing configuration is applied.</param>
+    /// <param name="withMetrics">An optional delegate to configure the metrics provider. If null, default metrics configuration is applied.</param>
+    public static void AddOpenTelemetryProvider(this ITestApplicationBuilder builder, Action<TracerProviderBuilder>? withTracing = null, Action<MeterProviderBuilder>? withMetrics = null)
+        => ((TestApplicationBuilder)builder).Telemetry.AddOpenTelemetryProvider(serviceProvider =>
+        {
+            ((ServiceProvider)serviceProvider).AddService(new OpenTelemetryPlatformService());
+            return new OpenTelemetryProvider(withTracing, withMetrics);
+        });
+
+    /// <summary>
+    /// Enables instrumentation for the Microsoft Testing Platform by adding its activity source to the specified tracer
+    /// provider builder.
+    /// </summary>
+    /// <remarks>Use this method to collect telemetry from components that emit activities under the
+    /// "Microsoft.Testing.Platform" source. This is typically required to enable distributed tracing for operations
+    /// performed by the Microsoft Testing Platform.</remarks>
+    /// <param name="builder">The tracer provider builder to which the Microsoft Testing Platform activity source will be added.</param>
+    /// <returns>The tracer provider builder with the Microsoft Testing Platform activity source configured for instrumentation.</returns>
+    public static TracerProviderBuilder AddTestingPlatformInstrumentation(this TracerProviderBuilder builder)
```

Comment:
Not sure of the name. Should this be `AddMicrosoftTestingPlatformInstrumentation`?